go/service/gregor.go

```diff
@@ -1016,36 +1043,88 @@ func (g *gregorHandler) auth(ctx context.Context, cli rpc.GenericClient) (err er
 	return nil
 }
 
+func (g *gregorHandler) isReachable() bool {
+	ctx := context.Background()
+	timeout := g.G().Env.GetGregorPingTimeout()
```

Comment:
10 seconds, I could separate them if you think we should go even shorter on the `Dial` one.